unittests/test_schedulers.py

```diff
@@ -544,116 +579,257 @@ def setUp(self):
+            self.prepare_job()
+
+    def test_negative_flex_alloc_tasks(self):
+        self.testjob._flex_alloc_tasks = '-4'
+        self.testjob._sched_access = ['--constraint=f1']
+        with self.assertRaises(JobError):
+            self.prepare_job()
+
+    def test_sched_access_idle(self):
+        self.testjob._flex_alloc_tasks = 'idle'
+        self.testjob._sched_access = ['--constraint=f1']
+        self.prepare_job()
+        self.assertEqual(self.testjob.num_tasks, 8)
+
+    def test_constraint_idle(self):
+        self.testjob._flex_alloc_tasks = 'idle'
+        self.testjob.options = ['--constraint=f1']
+        self.prepare_job()
+        self.assertEqual(self.testjob.num_tasks, 8)
+
+    def test_partition_idle(self):
+        self.testjob._flex_alloc_tasks = 'idle'
+        self.testjob._sched_partition = 'p2'
+        with self.assertRaises(JobError):
+            self.prepare_job()
+
+    def test_valid_constraint_short(self):
         self.testjob.options = ['-C f1']
         self.prepare_job()
-        self.assertEqual(self.testjob.num_tasks, expected_num_tasks)
+        self.assertEqual(self.testjob.num_tasks, 12)
+
+    def test_valid_constraint_long(self):
+        self.testjob.options = ['--constraint=f1']
+        self.prepare_job()
+        self.assertEqual(self.testjob.num_tasks, 12)
 
-    def test_valid_multiple_constraints(self, expected_num_tasks=4):
-        self.testjob._sched_reservation = 'Foo'
+    def test_valid_multiple_constraints(self):
         self.testjob.options = ['-C f1 f3']
         self.prepare_job()
-        self.assertEqual(self.testjob.num_tasks, expected_num_tasks)
+        self.assertEqual(self.testjob.num_tasks, 4)
 
-    def test_valid_partition(self, expected_num_tasks=8):
-        self.testjob._sched_reservation = 'Foo'
+    def test_valid_partition_cmd(self):
         self.testjob._sched_partition = 'p2'
         self.prepare_job()
-        self.assertEqual(self.testjob.num_tasks, expected_num_tasks)
+        self.assertEqual(self.testjob.num_tasks, 8)
 
-    def test_valid_multiple_partitions(self, expected_num_tasks=4):
-        self.testjob._sched_reservation = 'Foo'
-        self.testjob.options = ['-p p1 p2']
-        if expected_num_tasks:
-            self.prepare_job()
-            self.assertEqual(self.testjob.num_tasks, expected_num_tasks)
-        else:
-            self.assertRaises(JobError, self.prepare_job)
+    def test_valid_partition_short(self):
+        self.testjob.options = ['-p p2']
+        self.prepare_job()
+        self.assertEqual(self.testjob.num_tasks, 8)
+
+    def test_valid_partition_long(self):
+        self.testjob.options = ['--partition=p2']
+        self.prepare_job()
+        self.assertEqual(self.testjob.num_tasks, 8)
 
-    def test_valid_constraint_partition(self, expected_num_tasks=4):
-        self.testjob._sched_reservation = 'Foo'
+    def test_valid_multiple_partitions(self):
+        self.testjob.options = ['--partition=p1 p2']
```

Comment:
I don't think this is a valid Slurm option.

Reply:
Indeed when multiple partitions are needed, they have to be separated by a comma.

Reply:
Applied the suggested change